pages/login.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import Head from 'next/head';
+import Root from '../components/common/Root';
+import Footer from '../components/common/Footer';
+import commerce from '../lib/commerce';
+
+class LoginPage extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: '',
+      isError: false,
+      message: [],
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.loginCustomer = this.loginCustomer.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      email: event.target.value
+    });
+  }
+
+  /**
+   * Send login email
+   *
+   * @param {Event} e
+   */
+  loginCustomer(e) {
+    e.preventDefault();
+
+    const userEmail = this.state.email
+    // Reset messaging.
+    this.setState({
+      isError: false,
+      message: []
+    });
+
+    return commerce.customer.login(userEmail, window.location.href).then(() => {
+      this.setState({
+        isError: false,
+        message: [
+          'If that email address exists in our system, we\'ve just sent you a link to continue logging in!'
+        ]
+      });
+    })
+    .catch((error)=>{
+      this.setState({
+        isError: true,
+        message: error.data.error.errors.email
+      });
+    });
+  }
+
+  renderAlert() {
+    // Generate alert message as either list or single line.
+
+    if (this.state.message.length > 1) {
+      return <div className={`alert ${this.state.isError ? "alert-danger" : "alert-success"}`}>
+        <ul className="text-left mb-0">
+          {this.state.message.map((copy) => <li key={copy}>{copy}</li>)}
+        </ul>
+      </div>;
+    };
+
+    if (this.state.message.length === 1) {
+      return <div className={`alert ${this.state.isError ? "alert-danger" : "alert-success"}`}>
+        {this.state.message}
+      </div>;
+    };
+  }
```

Comment:
```suggestion
  renderAlert() {
    const { isError, message } = this.state;
    
    if (!message) {
      return null;
    }
    
    // Generate alert message as either list or single line.
    const alertMessage = typeof message === 'string'
      ? message
      : (
        <ul className="text-left m-0">
          { message.map((copy) => <li key={copy}>{copy}</li>) }
        </ul>
      );
    
    return (
      <div className={`alert ${isError ? 'alert-danger' : 'alert-success'}`}>
        { alertMessage }
      </div>
    );
  }
```

Suggestion for tidying this method up a bit

Reply:
Updated, let me know if this is better.